invenio_records_resources/records/systemfields/pid.py

```diff
@@ -180,3 +183,69 @@ def dump_obj(field, record, pid):
 
         # Set ID on desired dictionary key.
         field.set_dictkey(record, pid.pid_value)
+
+
+class ModelPIDFieldContext(PIDFieldContext):
+    """Context for ModelPIDField.
+
+    This class implements the class-level methods available on a PIDField. I.e.
+    when you access the field through the class, for instance:
+
+    .. code-block:: python
+
+        Record.pid.resolve('...')
+        Record.pid.session_merge(record)
+    """
+
+    def resolve(self, pid_value, registered_only=True):
+        """Resolve identifier."""
+        resolver = self.field._resolver_cls(
+            self._record_cls, self.field.model_field_name
+        )
+        pid, record = resolver.resolve(pid_value)
+        self.field._set_cache(record, pid)
+
+        return record
+
+    def create(self, record):
```

Comment:
Needed so the `PIDComponent` can create it. I am slightly puzzled this is needed. In my opinion, the `pid` attribute should be set on `invenio-records` when the "dict" is initialize (`RecordBase` constructor), removing the need for a PIDComponent in the first place. However, it does not seem to be being set (`ModelField` should do this right?)